### PR TITLE
Search specific group of names, gender and event location

### PR DESCRIPTION
**Pull request by euanmillar**
Currently the search handler automatically searches all names.

The chatbot mediator search will search by a group of specific names: child, mother & father.  So I need to amend the search to be able to send an optional object of specific names. 

I also need to add gender and event location as parameters to satisfy the design.

### CODE REVIEW

File: packages/search/src/features/search/service.ts [hunk 1]
```diff
@@ -26,29 +50,58 @@ export const searchComposition = async (params: ISearchQuery) => {
...
-        trackingId,
-        contactNumber,
-        registrationNumber,
-        name,
-        applicationLocationId,
-        createdBy,
-        { event, status, type }
-      ),
-      sort: [{ dateOfApplication: sort }]
+  if (ageCheck === 'TRUE' && nameCombinations.length > 0) {
```
Comment (mushrafulhoque-dsi) [inline packages/search/src/features/search/service.ts]: Just to be on safe side, we should do: `ageCheck.toUpperCase() === 'TRUE'`

File: packages/search/src/features/search/utils.ts [hunk 2]
```diff
@@ -167,3 +201,146 @@ export function queryBuilder(
     }
   }
 }
+
+export function combinationQueryBuilder(
```
Comment (mushrafulhoque-dsi) [inline packages/search/src/features/search/utils.ts]: Can we merge `queryBuilder` and `combinationQueryBuilder` function into one ? Most of the props are same in these 2 functions.

File: packages/search/src/features/registration/birth/handler.test.ts [hunk 3]
```diff
@@ -137,107 +136,5 @@ describe('Verify handlers', () => {
 
       expect(res.statusCode).toBe(500)
     })
-
-    it('should return status code 200 if the composition indexed correctly', async () => {
```
[the remark this answers is not in the document]
Comment (euanmillar) [reply packages/search/src/features/registration/birth/handler.test.ts]: I must have forgotten to put this back.

File: packages/search/src/features/fhir/fhir-utils.ts [hunk 4]
```diff
@@ -78,6 +78,25 @@ export function findEntry(
   return findEntryResourceByUrl(reference, bundleEntries)
 }
 
+export async function findLocation(
```
[the remark this answers is not in the document]
Comment (euanmillar) [reply packages/search/src/features/fhir/fhir-utils.ts]: Agreed

File: packages/search/src/features/search/service.ts (hunk 1 above)
Comment (euanmillar) [reply packages/search/src/features/search/service.ts]: Agreed

File: packages/search/src/features/search/utils.ts (hunk 2 above)
Comment (euanmillar) [reply packages/search/src/features/search/utils.ts]: Will look into this.

File: packages/search/src/features/registration/birth/handler.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (euanmillar) [reply packages/search/src/features/registration/birth/handler.test.ts]: Looks like I moved it to here with more fetch responses required.  packages/search/src/features/registration/birth/composition.test.ts

**Pull request merged**